projects/go-lib/src/lib/components/go-datepicker/go-datepicker.component.ts

```diff
@@ -54,6 +54,8 @@ export class GoDatepickerComponent implements OnDestroy, OnInit {
     this.subscription = this.control.valueChanges.subscribe((value: Date) => {
       if (!value && this.selectedDate) {
         this.control.setErrors([{ message: 'format is invalid' }]);
+      } else {
```

Comment:
I think we want `else if (value)` here. The reason is the control could be set to null by the datePicked method, because it's out of range, and we don't want to clear the field if they enter an invalid date, just show them it's not valid.

Reply:
Ah yes, good point. I've made the change.